### PR TITLE
Denormalize thread meta data (message & reaction counts)

### PR DESCRIPTION
**Pull request by mxstbr**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- api

**Run database migrations (delete if no migration was added)**
YES

**Related issues (delete if you don't know of any)**
Ref #3718

This is a pain, ugh, but unfortunately a necessary one. This adds a migration which adds denormalized message and reaction counts to the thread object, and tries to keep them in sync with the actual counts by plugging into their models. Please review carefully!

### CODE REVIEW

**Comment by mxstbr** [comment]
Not sure why these e2e tests keep failing, it all works fine locally 🤔

**Comment by mxstbr** [comment]
@brianlovin we should ship this to remove some load from loading threads from the db!

**Comment by brianlovin** [comment]
Merged alpha and fixed a couple merge conflicts.

Question: why should we even denormalize the reaction counts on threads if we're not going to use them yet? In the thread.reactions resolver we still have to db query every thread reaction anyways in order to return the `hasReacted` boolean. So maybe that's unnecessary? 

Denormalizing the message counts seems like a good step regardless!

**Comment by brianlovin** [comment]
LOL my commit deleted the whole code base. One sec.

**Comment by brianlovin** [comment]
Also wanna double check my merge in? https://github.com/withspectrum/spectrum/pull/3726/commits/72fd522e1e141a201a584e1880ea25ce69aaa56d

The conflict was that we made it so that thread reaction notifications only get sent to non-authors :)

File: shared/types.js [hunk 1]
```diff
@@ -275,6 +275,8 @@ export type DBThread = {
   attachments?: Array<DBThreadAttachment>,
   edits?: Array<DBThreadEdits>,
   watercooler?: boolean,
+  messageCount: number,
+  reactionCount: number,
```
Comment (brianlovin) [inline shared/types.js:279]: This field will technically exist after this branch merges, but it's not being used - that's what i mean by my above comment

**Comment by brianlovin** [comment]
Migration + functionality all working as expected locally though, so great work :)

**Comment by mxstbr** [comment]
> Question: why should we even denormalize the reaction counts on threads if we're not going to use them yet? In the thread.reactions resolver we still have to db query every thread reaction anyways in order to return the hasReacted boolean. So maybe that's unnecessary?

Not necessarily, that db query can then be specific to `hasReacted`, something like `hasReactedToThread(userId, threadId)`. No need to send all the reactions over the wire! I reckon we should also do that in this PR

**Comment by brianlovin** [comment]
> No need to send all the reactions over the wire! I reckon we should also do that in this PR

We don't send reactions over the wire anyways. Tbh though we probably want to start doing this in the future so that if someone hovers over a reaction count we can fetch the names of all the people who have reacted to a message. Right now it's frustrating to not know who liked your stuff :P

Anyways - since that's not needed now, and it doesn't hurt to have the denormalized reaction count ship with this migration anyways, I'm going to go ahead and merge + deploy

**Review by brianlovin — APPROVED** [review]

**Comment by mxstbr** [comment]
> We don't send reactions over the wire anyways.

I mean the wire between db and server, sorry for not being clear!

**Comment by brianlovin** [comment]
Ah, of course. FYI this is deployed! I just finished the migration and aliased the api

**Comment by brianlovin** [comment]
Currently confirming everything is working in prod

**Pull request merged**